.buildkite/build-and-test.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e
```

Comment:
FYI, in [another script](https://github.com/wordpress-mobile/WordPress-iOS/pull/16806#discussion_r666077522), @AliSoftware and I have been discussing using `#!/bin/sh -euo pipefail`. The context is different from here, because that script is meant to run as a build phase from Xcode and uses `sh` instead of `bash` (for no particular reason) but might be worth considering.

Although, I did a quick google around and have found mixed opinions on whether it is a best practice or not. That's not to say that it definitely isn't, only that I don't have much experience in the matter and I'm still trying to figure it out 😄

Reply:
By the way, is there a particular reason why you chose `bash` instead of `sh` this and the other scripts?

Reply:
> By the way, is there a particular reason why you chose bash instead of sh this and the other scripts?

We have had this discussion before in Slack - this might have been before you and @AliSoftware  joined the team. Basically, I can't think of a good reason to pick `sh` over `bash` (assuming `bash` is available) as we'd just be limiting ourselves with it.

Reply:
Fine by me 😄 I'll adopt this a standard moving forward 👍 Thanks for the extra context.

Reply:
> FYI, in another script, @AliSoftware and I have been discussing using #!/bin/sh -euo pipefail.

To be clear, the thing about `set -euo pipefail` vs setting `-euo pipefail` on the shebang… is a bit like tabs vs spaces: I don't think there's a strong rationale for one over the other, and ultimately it doesn't matter and it's ok to go with whatever you like (I certainly don't want to give the wrong impression I wanted to start a debate over this 😅). Both are equally fine and it isn't a big deal which one we choose to use.